src/org/zaproxy/zap/extension/websocket/manualsend/ManualWebSocketSendEditorDialog.java

```diff
@@ -261,5 +322,130 @@ public void addSeparator() {
 		public void addEndButton(JButton button) {
 			messagePanel.addOptions(button, HttpPanel.OptionsLocation.END);
 		}
+		
+		private JButton getBtnReopen() {
+			if (btnReopen == null) {
+				btnReopen = new JButton();
+				btnReopen.setIcon(REOPEN_ICON);
+				btnReopen.setEnabled(true);
+				btnReopen.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_hint"));
+				btnReopen.addActionListener((ActionEvent e) -> {
+					isOnConnectingProgress(true);
+					String newSecWebSocketKey = null;
+					
+					if(!wsUiHelper.isAlwaysGenerate()){
+						newSecWebSocketKey = wsUiHelper.getWebSocketKey();
+					}
+					
+					ReEstablishConnection reEstablishConnection = new ReEstablishConnection(newSecWebSocketKey);
+					
+					reEstablishConnection.addPropertyChangeListener(propertyChangeEvent -> {
+						if((propertyChangeEvent.getNewValue() == SwingWorker.StateValue.DONE)){
+							isOnConnectingProgress(false);
+							int channelId;
+							try {
+								channelId = reEstablishConnection.get();
+								if(channelId < 0){
+									if(channelId == ReEstablishConnection.RETRIVING_HANDSHAKE_ERROR){
+										//TODO: This message should be replaced when we are able to open websocket connection.
+										View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.retrieve"));
+									}else if(channelId == ReEstablishConnection.REQUEST_OUT_OF_SCOPE){
+										View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.out_of_scope"));
+									}else if(channelId == ReEstablishConnection.IOEXCEPTION){
+										View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.unable_reopen"));
+									}
+								}else{
+									wsUiHelper.setSelectedChannelId(channelId);
+									wsUiHelper.requestFocusChannelComboBox();
+								}
+							} catch (Exception ex){
+								LOGGER.error("The background swing worker was interrupted when that was tried to re-establish a websocket connection",ex);
```

Comment:
This could be just `An error occurred when trying to re-establish a websocket connection`, it might not be an interrupt.